.github/workflows/CI.yml

```diff
@@ -25,115 +38,67 @@ jobs:
-        if: matrix.rust == 'nightly'
-      # not nightly
-      - name: "gdkx11: tests"
-        run: xvfb-run --auto-servernum cargo test --manifest-path gdkx11/Cargo.toml --features "v3_24"
-        if: matrix.rust != 'nightly'
-      - name: "gdkx11-sys: tests"
-        run: xvfb-run --auto-servernum cargo test --manifest-path gdkx11/sys/Cargo.toml --features "v3_24"
-        if: matrix.rust != 'nightly'
-      - name: "gdkx11: build"
-        run: cargo build --manifest-path gdkx11/Cargo.toml --features "v3_24"
-        if: matrix.rust != 'nightly'
-      # gio
-      # nightly
-      - name: "gio: tests nightly"
-        run: xvfb-run --auto-servernum cargo test --manifest-path gio/Cargo.toml --all-features
-        if: matrix.rust == 'nightly'
-      - name: "gio: build nightly"
-        run: cargo build --manifest-path gio/Cargo.toml --all-features
-        if: matrix.rust == 'nightly'
-      # not nightly
-      - name: "gio: tests"
-        run: xvfb-run --auto-servernum cargo test --manifest-path gio/Cargo.toml --features "v2_66"
-        if: matrix.rust != 'nightly'
-      - name: "gio: build"
-        run: cargo build --manifest-path gio/Cargo.toml --features "v2_66"
-        if: matrix.rust != 'nightly'
-      # glib
-      # nightly
-      - name: "glib: tests nightly"
-        run: xvfb-run --auto-servernum cargo test --manifest-path glib/Cargo.toml --all-features
-        if: matrix.rust == 'nightly'
-      - name: "glib: build nightly"
-        run: cargo build --manifest-path glib/Cargo.toml --all-features
+      - name: tests
+        run: xvfb-run --auto-servernum cargo test --manifest-path ${{ matrix.conf.name }}/Cargo.toml --features "${{ matrix.conf.features }}"
+        if: matrix.rust != 'nightly'
+      - name: tests sys
+        run: xvfb-run --auto-servernum cargo test --manifest-path ${{ matrix.conf.name }}/sys/Cargo.toml --features "${{ matrix.conf.features }}"
+        if: matrix.rust != 'nightly' && matrix.conf.test_sys
+      - name: build
+        run: cargo build --manifest-path ${{ matrix.conf.name }}/Cargo.toml --features "${{ matrix.conf.features }}"
+        if: matrix.rust != 'nightly'
+
+      # rustdoc-stripper
+      - name: doc add
+        run: cargo build --manifest-path ${{ matrix.conf.name }}/Cargo.toml --features "dox,embed-lgpl-docs"
+        if: matrix.rust == 'nightly' && matrix.conf.name != 'glib'
+      - name: doc removal
+        run: cargo build --manifest-path ${{ matrix.conf.name }}/Cargo.toml --features "purge-lgpl-docs"
+        if: matrix.rust == 'nightly' && matrix.conf.name != 'glib'
+      - name: glib doc check
+        run: cargo build --manifest-path ${{ matrix.conf.name }}/Cargo.toml --features "dox"
+        if: matrix.rust == 'nightly' && matrix.conf.name == 'glib'
+      - name: check diff
+        run: git diff -R --exit-code
         if: matrix.rust == 'nightly'
+
+      - uses: bcomnes/cleanup-xvfb@v1
+
+  build-others:
```

Comment:
Why do we need to split `build-others` to its own job? with a couple of tweaks it can be part of the main build